assets/src/edit-story/components/devTools/devTools.js

```diff
@@ -0,0 +1,210 @@
+            newElement.resource.poster =
+              'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAYAAABytg0kAAAAGElEQVQYV2NkYGD4////fwZGEMnIyMgAAEvqB/6wfXLSAAAAAElFTkSuQmCC';
+
+            const videoEl = document.getElementById(`video-${newElement.id}`);
+            if (videoEl) {
+              videoEl.load();
+            }
+          }
+          if (element.type === 'image') {
+            newElement.resource.mimeType = 'image/png';
+            newElement.resource.src =
+              'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAIAAAACCAYAAABytg0kAAAAGElEQVQYV2NkYGD4////fwZGEMnIyMgAAEvqB/6wfXLSAAAAAElFTkSuQmCC';
+          }
+        }
+        return newElement;
+      }),
+    })),
+  };
+};
+
+function DevTools() {
+  const [isVisible, setIsVisible] = useState(false);
+  const [isBase64, setIsBase64] = useState(false);
+  const [isDummyResources, setIsDummyResources] = useState(false);
+  const { showSnackbar } = useSnackbar();
+  const textareaRef = useRef();
+  const {
+    state: { reducerState },
+    actions: { restore },
+  } = useStory();
+  // This will not work here
+  // const {
+  //   state, // stale data
+  //   internal: { restore }, // will not update
+  // } = useStoryReducer();
+
+  const {
+    pages,
+    current,
+    selection,
+    story: { stylePresets },
+  } = reducerState;
+  const reducerStateSlice = {
+    current,
+    selection,
+    story: { stylePresets },
+    pages,
+  };
+  const storyData = isDummyResources
+    ? replaceResourcesWithDummy(reducerStateSlice)
+    : reducerStateSlice;
+
+  const toggleDummyResources = () => setIsDummyResources((v) => !v);
+  const toggleBase64 = () => setIsBase64((v) => !v);
+  const toggleVisible = () => setIsVisible((v) => !v);
+  const copyToClipboard = () => {
+    textareaRef.current.select();
+    document.execCommand('copy');
+    textareaRef.current.setSelectionRange(0, 0);
+    showSnackbar({ message: 'Copied to clipboard', timeout: 1200 });
+  };
+  const loadFromInput = () => {
+    let input = output;
+    try {
+      input = window.atob(input);
```

Comment:
This is a bit rough. Can we check that a first non-whitespace char is a `{` to indicate JSON?

Reply:
Or we can do try/catch on `JSON.parse` but isn't that the same?

Reply:
Hmm. Nevermind actually. These two formats really never overlap. We can just keep the way it is.